lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxImpl.scala

```diff
@@ -1,16 +1,27 @@
 package org.ergoplatform.appkit.impl
 
 import org.ergoplatform.ErgoBoxCandidate
-import org.ergoplatform.appkit.{ErgoId, ErgoToken, OutBox}
+import org.ergoplatform.appkit.{ErgoToken, ErgoValue, Iso, OutBox}
 import scorex.util.ModifierId
+import sigmastate.Values
 
-class OutBoxImpl(_ctx: BlockchainContextImpl,
-                 _ergoBoxCandidate: ErgoBoxCandidate) extends OutBox {
+import java.util
+
+class OutBoxImpl(_ctx: BlockchainContextImpl, _ergoBoxCandidate: ErgoBoxCandidate) extends OutBox {
   override def getValue: Long = _ergoBoxCandidate.value
 
   override def getCreationHeight: Int = _ergoBoxCandidate.creationHeight
 
-  override def token(id: ErgoId): ErgoToken = null
+  override def getTokens: util.List[ErgoToken] = Iso.isoTokensListToPairsColl.from(_ergoBoxCandidate.additionalTokens)
+
+  override def getRegisters: util.List[ErgoValue[_]] = {
+    val registers = _ergoBoxCandidate.additionalRegisters.values.toIndexedSeq
+    Iso.JListToIndexedSeq(Iso.isoErgoValueToSValue).from(registers)
+  }
+
+  override def getBytes: Array[Byte] = _ergoBoxCandidate.bytesWithNoRef
```

Comment:
I suggest to rename the method `getBytesWithoutRef` to be consistent with Sigma (I don't like the original `bytesWithNoRef`, but it is there historically and here we can change it a bit).

And also add `getBytes` method which will return the full bytes.

Reply:
I am not sure how to get the full bytes for ErgoBoxCandidate, the only method in sigmastate that I can see is bytesWithNoRef. I am not sure that it is possible to get the full bytes if there is no transaction reference data, so should I be inputting placeholder values to get the full bytes?

Reply:
Let's leave getBytes for another PR.